buildSrc/src/main/groovy/com.flipkart.varadhi.java-library-conventions.gradle

```diff
@@ -4,4 +4,29 @@ plugins {
 
     // Apply the java-library plugin for API and implementation separation.
     id 'java-library'
+
+    // Apply the maven-publish to allow deploying to maven artifactories
+    id 'maven-publish'
+}
+
+
+publishing {
+    publications {
+        maven(MavenPublication) {
+            groupId = "${projectGroupId}"
+            artifactId = "${project.name}"
+            version = "${version}"
```

Comment:
How will this result in snapshot or unstable version being published ?

Reply:
won't be published unless gradlew publish is run
I have published SNAPSHOT version, but pushed changes have omitted SNAPSHOT since the docker build was failing

Reply:
I think in code it should be SNAPSHOT by default. Why docker build was failing ?
(what I meant, is adding publishing to common convention will make that target visible to all submodules. Someone running publish at root, will execute it for all modules which we can avoid).

Reply:
docker build is hardcoded to use specific verison
I'll fix that